[lit-next] Unable to use Lit 1 and 2 in the same TypeScript application

# Description

It does not seem possible to compile TypeScript applications where both Lit 1 and 2 are used

# Steps to Reproduce

1. Write this code

```ts
import {html as lit1html} from 'lit-element';
import {html as lit2html} from 'lit';

console.log(lit1html);
console.log(lit2html);
```

## Live Reproduction Link
```
git clone https://github.com/Artur-/test-lit1-lit2-ts
cd test-lit1-lit2-ts
npm run build
```

## Expected Results
It builds

## Actual Results
```
> lit-element-starter-ts@0.0.0 build
> tsc

node_modules/@lit/reactive-element/reactive-element.d.ts:24:11 - error TS2451: Cannot redeclare block-scoped variable 'JSCompiler_renameProperty'.

24     const JSCompiler_renameProperty: <P extends PropertyKey>(prop: P, _obj: unknown) => P;
             ~~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/lit-element/lib/updating-element.d.ts:15:9
    15     var JSCompiler_renameProperty: <P extends PropertyKey>(prop: P, _obj: unknown) => P;
               ~~~~~~~~~~~~~~~~~~~~~~~~~
    'JSCompiler_renameProperty' was also declared here.

node_modules/lit-element/lib/updating-element.d.ts:15:9 - error TS2451: Cannot redeclare block-scoped variable 'JSCompiler_renameProperty'.

15     var JSCompiler_renameProperty: <P extends PropertyKey>(prop: P, _obj: unknown) => P;
           ~~~~~~~~~~~~~~~~~~~~~~~~~

  node_modules/@lit/reactive-element/reactive-element.d.ts:24:11
    24     const JSCompiler_renameProperty: <P extends PropertyKey>(prop: P, _obj: unknown) => P;
                 ~~~~~~~~~~~~~~~~~~~~~~~~~
    'JSCompiler_renameProperty' was also declared here.


Found 2 errors.
```

--- a/packages/reactive-element/CHANGELOG.md
+++ b/packages/reactive-element/CHANGELOG.md
@@ -33,6 +33,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - (Since 1.0.0-pre.1) Removed `willUpdate` from `ReactiveController`.
 - (Since 1.0.0-pre.1) Renamed `Controller`'s `dis/connectedCallback` methods.
 - (Since 1.0.0-pre.1) Renamed `Controller` to `ReactiveController`.
+- Made JSCompiler_renameProperty block scoped so that it's inlined in the Terser prod build. Closure should compile from the development build, or after a custom TypeScript compilation.
 
 ## [1.0.0-pre.1] - 2020-12-16
 

--- a/packages/reactive-element/src/reactive-element.ts
+++ b/packages/reactive-element/src/reactive-element.ts
@@ -75,21 +75,11 @@ if (DEV_MODE) {
  * alias this function, so we have to use a small shim that has the same
  * behavior when not compiling.
  */
-window.JSCompiler_renameProperty = <P extends PropertyKey>(
+/*@__INLINE__*/
+const JSCompiler_renameProperty = <P extends PropertyKey>(
   prop: P,
   _obj: unknown
 ): P => prop;
-
-declare global {
-  const JSCompiler_renameProperty: <P extends PropertyKey>(
-    prop: P,
-    _obj: unknown
-  ) => P;
-
-  interface Window {
-    JSCompiler_renameProperty: typeof JSCompiler_renameProperty;
-  }
-}
 
 /**
  * Converts property values to and from attribute values.
@@ -1160,7 +1150,9 @@ if (DEV_MODE) {
   // Default warning set.
   ReactiveElement.enabledWarnings = ['change-in-update'];
   const ensureOwnWarnings = function (ctor: typeof ReactiveElement) {
-    if (!ctor.hasOwnProperty('enabledWarnings')) {
+    if (
+      !ctor.hasOwnProperty(JSCompiler_renameProperty('enabledWarnings', ctor))
+    ) {
       ctor.enabledWarnings = ctor.enabledWarnings!.slice();
     }
   };

